src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/get_vehicle_data_request.cc

```diff
@@ -170,6 +170,8 @@ void GetVehicleDataRequest::on_event(const event_engine::Event& event) {
           }
         }
 
+        MessageHelper::RemoveEmptyMessageParams(message[strings::msg_params]);
```

Comment:
Perhaps this is out of the scope of this issue, but it seems odd to me that a warning isn't returned if only some of the requested data is received from the HMI.
Ex. request:
```
{
  vin: true,
  fuelRange: true
}
```

HMI response:
```
   fuelRange: [{}],
   vin: "fakevin"
```

Mobile response:
```
{
  success: true,
  resultCode: "SUCCESS",
  vin: "fakevin"
}
```

I would expect:
```
{
  success: true,
  resultCode: "WARNINGS",
  info: "Some vehicle data items could not be retrieved",
  vin: "fakevin"
}
```

Reply:
@jacobkeeler agree, it makes more sense to return WARNINGS in that case.
I updated function to return the amount of parameters removed, so if at least one param has been removed, SDL will return WARNINGS
Implemented in https://github.com/smartdevicelink/sdl_core/pull/3751/commits/8d8002808ba96efa3fc9a29392ca2261392f4704
It is out of scope of that fix but I believe it can be merged all together if you are ok with that